### DISCUSSION TITLE
how to incorporate information from an application

### DISCUSSION THREAD

**Discussion by richy1996** [hootnot/saxo_openapi#31] [Q&A]
So once you start moving towards live trading, it seems you have to create an application according to [this](https://www.developer.saxo/openapi/learn/writing-an-openapi-application). So I created one for the sim environment already. Now I have this information
```python
{
  "AppName": "blabla",
  "AppKey": ";lkjasdlkjasdf",
  "AuthorizationEndpoint": "https://sim.logonvalidation.net/authorize",
  "TokenEndpoint": "https://sim.logonvalidation.net/token",
  "GrantType": "Code",
  "OpenApiBaseUrl": "https://gateway.saxobank.com/sim/openapi/",
  "RedirectUrls": [
    "http://bla.bla.com"
  ],
  "AppSecret": "asdfl;kjasdlfkj"
}
```

How do I incorporate this information? I'm guessing i'll need to modify something when I create the client here
```python
client = API(access_token=token)
# and fetch our account information
ai = account_info(client)
```

I've tried a few different things with no luck so far. Thx for any help.

**Comment by hootnot** [comment]
You need to set up the OAuth2 flow.
You register your application. The information you get from that  is what you need in the OAuth2 config.
If you then login, answer the dialogs, then you get  OAuth2  going and you will have a token that you can use for you SAXO application. That token expires every ... 20 min. I think. Using the refresh token you can fetch a new token.  
If you keep that going, you will always have a token available for the API.

In your application you need the `live` environment, so:
```
client = API(access_token=token, environment="live")
```

I don't know on what environment you are. Manage OAuth2 using docker is an option
